File: Turculescu_Samuel/Week_5/Domain/Entities/Order.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace Persistence.EfCore
{
    public partial class Order
    {
        public Order()
        {
            OrderItems = new List<OrderItem>();
        }

        public int Id { get; set; }
        public int ClientId { get; set; }
        public int RestaurantId { get; set; }
        public uint TableNumber { get; set; }
        public DateTime DateTimePlaced { get; set; }
        public decimal TotalPrice { get; set; }
        public string OrderStatus { get; set; }
        public TimeSpan PreparationTime { get; set; }
        public string PaymentStatus { get; set; }

        public virtual Client Client { get; set; }
        public virtual Restaurant Restaurant { get; set; }
        public virtual List<OrderItem> OrderItems { get; set; }
    }
}

File: Turculescu_Samuel/Week_8/order-and-pay-employee/src/models/order.ts
import { OrderItem } from "./../models/orderItem";
import { Restaurant } from "./../models/restaurant";

export type Order = {
    id: string,
    clientId: string,
    restaurantId: string,
    tableNumber: number,
    dateTimePlaced: Date,
    totalPrice: number,
    orderStatus: string,
    preparationTime: TimeRanges,
    paymentStatus: string
    
    // client: Client,
    restaurant: Restaurant,
    orderItems: OrderItem[]
}